File: backend/backend/src/SignalingComponent/Common/Api/DTOs/IceCandidateMessage.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.SignalingComponent.Common.Api.DTOs;

public class IceCandidateMessage : ITypedMessage
{
    public static string TypeString => "ice-candidate";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("remoteToken")]
    public required string RemoteToken { get; set; }

    [JsonPropertyName("iceCandidate")]
    public JsonElement IceCandidate { get; set; }

}

File: frontend/src/types/rtc/IceCandidateMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class IceCandidateMessage implements ITypedMessage {
    public readonly type = MessageType.ICE_CANDIDATE;
    public msg: {
        remoteToken: string;
        iceCandidate: RTCIceCandidate;
    };

    public constructor(msg: {
        remoteToken: string;
        iceCandidate: RTCIceCandidate;
    }) {
        this.msg = msg;
    }
}
